core/src/main/java/org/geysermc/geyser/translator/inventory/item/ItemTranslator.java

```diff
@@ -149,97 +163,126 @@ private static ItemData.Builder translateToBedrock(GeyserSession session, Item j
+                continue;
+            }
+
+            // Declare the slot, e.g. "When in Main Hand"
             Component slotComponent = Component.text()
                     .resetStyle()
                     .color(NamedTextColor.GRAY)
                     .append(Component.newline(), Component.translatable("item.modifiers." + slot))
                     .build();
-            listTag.add(new StringTag("", MessageTranslator.convertMessage(slotComponent, language)));
+            lore.add(new StringTag("", MessageTranslator.convertMessage(slotComponent, language)));
 
+            // Then list all the modifiers when used in this slot
+            for (StringTag modifier : modifiers) {
+                lore.add(modifier);
+            }
+        }
 
-            for (Tag modifier : modifiersList) {
-                Map<String, Tag> modifierValue = (Map) modifier.getValue();
-                double amount;
-                if (modifierValue.get("Amount") instanceof IntTag intTag) {
-                    amount = (double) intTag.getValue();
-                } else if (modifierValue.get("Amount") instanceof DoubleTag doubleTag) {
-                    amount = doubleTag.getValue();
-                } else {
-                    continue;
-                }
-                if (amount == 0) {
-                    continue;
-                }
-                ModifierOperation operation = ModifierOperation.from((int) modifierValue.get("Operation").getValue());
-                String operationTotal;
-                if (operation == ModifierOperation.ADD) {
-                    if (modifierValue.get("Name").equals("knockback_resistance")) {
-                        amount *= 10;
-                    }
-                    operationTotal = decimalFormat.format(amount);
-                } else if (operation == ModifierOperation.ADD_MULTIPLIED || operation == ModifierOperation.MULTIPLY) {
-                    operationTotal = decimalFormat.format(amount * 100) + "%";
-                } else {
-                    continue;
-                }
-                if (amount > 0) {
-                    operationTotal = "+" + operationTotal;
-                }
+        displayTag.put(lore);
+        nbt.put(displayTag);
+    }
 
-                Component attributeComponent = Component.text()
-                        .resetStyle()
-                        .color(amount > 0 ? NamedTextColor.BLUE : NamedTextColor.RED)
-                        .append(Component.text(operationTotal), Component.text(" "), Component.translatable("attribute.name." + modifierValue.get("Name").getValue()))
-                        .build();
-                listTag.add(new StringTag("", MessageTranslator.convertMessage(attributeComponent, language)));
-            }
+    @Nullable
+    private static String attributeToLore(CompoundTag modifier, String language) {
+        double amount;
+        if (modifier.get("Amount") instanceof IntTag intTag) {
```

Comment:
A better implementation (working around the current OpenNBT) would be saving `modifier.get("Amount")` to a variable and checking for nullness and if `getValue()` is an instance of Number (if so, call `.doubleValue()`).